README-vault.md

```diff
@@ -0,0 +1,226 @@
+Vault module
+===================
+
+Description
+-----------
+
+The vault module allows to ensure presence and absence of vault and members of vaults.
+
+The vault module is as compatible as possible to the Ansible upstream `ipa_vault` module, and additionally offers to make sure that vault members, grups and owres are present or absent in a vault, and allow the archive, exclusion and retrieval of data from vaults.
+
+
+Features
+--------
+* Vault management
+
+
+Supported FreeIPA Versions
+--------------------------
+
+FreeIPA versions 4.4.0 and up are supported by the ipavault module.
+
+
+Requirements
+------------
+
+**Controller**
+* Ansible version: 2.8+
+
+**Node**
+* Supported FreeIPA version (see above)
+* KRA service must be enable
+
+
+Usage
+=====
+
+Example inventory file
+
+```ini
+[ipaserver]
+ipaserver.test.local
+```
+
+Example playbook to make sure vault is present:
+
+```yaml
+---
+- name: Playbook to handle vaults
+  hosts: ipaserver
+  become: true
+
+  tasks:
+  - ipavault:
+      ipaadmin_password: MyPassword123
+      name: symvault
+      username: admin
+      vault_password: MyVaultPassword123
+      description: A standard private vault.
+```
+
+Example playbook to make sure that a vault and its members are present:
+
+```yaml
+---
+- name: Playbook to handle vaults
+  hosts: ipaserver
+  become: true
+
+  tasks:
+  - ipasudocmdgroup:
+      ipaadmin_password: MyPassword123
+      name: symvault
+      username: admin
+      users: user01
+```
+`action` controls if the vault, data, member or owner will be handled. To add or remove members, set `action` to `member`. To add or remove owners, set `action` to `owner`. To add or remove data to the vault, set `action` to `data`.
```

Comment:
Why not using an owner parameter for owners? There is no need to add another action for this. Also `action: data` is not needed as `vault_data` is used at the same time.

Reply:
The use of `action: data` is for retrieving data stored in the vault, when `vault_data` is not present.

I agree with the usage of `owner` and `ownergruops`, which would make `action: owner` irrelevant.

Reply:
BTW, I'm not 'sold' in using `action: data`, that's what I came up with, but feel it could be better.

Reply:
Modified it by droping data retrieval.